core/src/main/java/org/elasticsearch/indices/recovery/RecoveryTarget.java

```diff
@@ -304,6 +304,7 @@ public void messageReceived(final RecoveryTranslogOperationsRequest request, fin
                 assert recoveryStatus.indexShard().recoveryState() == recoveryStatus.state();
                 try {
                     recoveryStatus.indexShard().performBatchRecovery(request.operations());
+                    channel.sendResponse(TransportResponse.Empty.INSTANCE);
```

Comment:
I would love to have those, but it will be a bigger rewrite - too many incoming dependencies. Next time we re-write something in this class...